### DISCUSSION TITLE
atomWithLocation triggers infinite renders with Preact Suspense

### DISCUSSION THREAD

**Discussion by maikeriva** [pmndrs/jotai#2094] [Bug report]
Hello, I am building a Preact app and I encountered an infinite re-render situation while using **atomWithLocation()** in combination with **Suspense**. Here's the code to reproduce:
```ts
import { FunctionComponent } from "preact";
import { Suspense } from 'preact/compat';
import { atom, useAtomValue } from "jotai";
import { atomWithLocation } from "jotai-location";

// NOTE: This triggers infinite re-rendering
const locationAtom = atomWithLocation()
// NOTE: This one instead does not
// const locationAtom = atom(window.location)

const atom1 = atom(async (get) => {
  console.log(`Evaluating atom1`)
  get(locationAtom)
  return 'something'
})

export const Content: FunctionComponent = () => {
  const location = useAtomValue(locationAtom)
  const value = useAtomValue(atom1)

  return <>
    <p>Infintely re-rendering 🙁</p>
  </>
}

export const App: FunctionComponent = () => {
  return <>
    <Suspense fallback={<p>Loading...</p>}>
      <Content />
    </Suspense>
  </>
}
```
From what I found out, this happens with the following conditions:
- We have an atom depending on atomWithLocation()
- Said atom getter is async
- We use Suspense

If we replace atomWithLocation with atom(window.location) the issue does not happen.

Not sure if the root of the issue lies in preact or jotai-location, but I thought that posting here could be a good starting point.

**Comment by dai-shi** [comment]
Doen't it happen with React's Suspense?

Maybe, you or someone need to dig into it and create a reproduction without jotai-location.
